src/olympia/devhub/tests/test_tasks.py

```diff
@@ -656,6 +658,12 @@ def test_webextension_webext_to_webext(self):
         upload.refresh_from_db()
 
         assert 'is_upgrade_to_webextension' not in upload.processed_validation
+        expected = ['validation', 'messages', 'webext_upgrade']
+        assert not any(
+            msg['id'] == expected
+            for msg in upload.processed_validation['messages'])
```

Comment:
Save upload.processed_validation in a variable so it's not being calculated numerous times?  (the property isn't cached afaik)
Same other in other places.